docs/ml_system_design_doc.md

```diff
@@ -0,0 +1,201 @@
+
+- Для MVP: Исследование и создание набора агентов для обработки ввода пользователя и интеграции его действий в мир и сюжет. Реализация базового набора команд и действий.
+- Для бейзлайна: Обработка ограниченного набора предопределенных команд.
+- Метрики качества: Процент успешно распознанных команд пользователя (не менее 85%), естественность ответов системы (оценка пользователями, не менее 7/10).
+- Риски: Ограниченное понимание естественного языка, шаблонные ответы. Планируем расширять набор понимаемых команд на основе тестирования.
+
+*Этап 7 - Интеграция и тестирование*
+
+- Для MVP и бейзлайна: Интеграция всех компонентов, проведение внутреннего тестирования, устранение ошибок и несогласованностей.
+- Метрики качества: Стабильность системы (не более 1 сбоя на 100 взаимодействий), общая оценка пользовательского опыта (не менее 7/10).
+- Риски: Интеграционные проблемы, несогласованность между агентами. Планируем поэтапное тестирование и итеративное улучшение.
+
+### 3. Подготовка пилота  
+  
+#### 3.1. Способ оценки пилота  
+
+- Тестовой группе из 20-30 пользователей будет предоставлен доступ к системе на 2 недели. Пользователи будут создавать персонажей и проходить сгенерированные приключения. После каждой сессии будут собираться количественные метрики (время сессии, количество взаимодействий) и качественная обратная связь через опросники. Также будут проведены интервью с частью пользователей для получения более детальной обратной связи.
+
+#### 3.2. Что считаем успешным пилотом  
+
+- Количественные метрики:
+  - Не менее 70% пользователей проводят в системе более 30 минут за сессию
+  - Не менее 60% пользователей возвращаются для повторных сессий
+  - Средняя оценка качества генерируемого контента не менее 7/10
+  - Средняя оценка удобства взаимодействия не менее 6/10
+- Качественные метрики:
+  - Положительные отзывы о целостности и увлекательности генерируемых миров и историй
+  - Желание пользователей увидеть дальнейшее развитие системы
+
+#### 3.3. Подготовка пилота  
+
+- Исходя из ожидаемых вычислительных затрат, для пилота будем использовать предобученные модели среднего размера с оптимизацией для быстрого инференса. Планируем эксперименты с бейзлайном для оценки необходимых вычислительных ресурсов и установления ограничений для MVP. По результатам экспериментов может потребоваться оптимизация архитектуры для обеспечения приемлемого времени отклика.
+
+### 4. Внедрение  
+  
+#### 4.1. Архитектура решения   
+
+- Система будет реализована как веб-приложение с клиент-серверной архитектурой:
+  - Фронтенд: React-приложение для взаимодействия с пользователем
+  - Бэкенд: Python-сервер с FastAPI для обработки запросов
+  - AMI (Главный агент): Ядро системы, которое напрямую взаимодействует с пользователем и координирует работу специализированных агентов 
+  - Мультиагентная система: Оптимальный набор специализированных агентов, определенный в процессе исследования и проектирования
```

Comment:
Do you plan to use any framework for developing agents? Or will it be custom-made?